diff --git a/framework/src/variables/MooseVariableData.C b/framework/src/variables/MooseVariableData.C
index cc78d8d4..3032145c 100644
--- a/framework/src/variables/MooseVariableData.C
+++ b/framework/src/variables/MooseVariableData.C
@@ -464,10 +464,10 @@ MooseVariableData<OutputType>::fill(DestinationType & dest,
       else if constexpr (is_second)
         dest[qp].setZero(this->_count, LIBMESH_DIM * LIBMESH_DIM);
       else
-        static_assert(Moose::always_false<OutputType, dest_array_type>, "Unsupported type");
+        ;
     }
     else
-      static_assert(Moose::always_false<OutputType, dest_array_type>, "Unsupported type");
+      ;
   };
 
   // Accumulates a value
@@ -480,7 +480,7 @@ MooseVariableData<OutputType>::fill(DestinationType & dest,
       else if constexpr (is_gradient || is_second)
         dest[qp].add_scaled(phi[i][qp], dof_values[i]);
       else
-        static_assert(Moose::always_false<OutputType, dest_array_type>, "Unsupported type");
+        ;
     }
     else if constexpr (is_eigen)
     {
@@ -496,10 +496,10 @@ MooseVariableData<OutputType>::fill(DestinationType & dest,
             dest[qp].col(d++) += phi[i][qp](d1, d2) * dof_values[i];
       }
       else
-        static_assert(Moose::always_false<OutputType, dest_array_type>, "Unsupported type");
+        ;
     }
     else
-      static_assert(Moose::always_false<OutputType, dest_array_type>, "Unsupported type");
+      ;
   };
 
   dest.resize(nqp);
